client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -151,7 +150,11 @@ const Checkout = React.createClass( {
 			return false;
 		}
 
-		if ( this.props.transaction.step.name === transactionStepTypes.SUBMITTING_WPCOM_REQUEST ) {
+		// Do not redirect if cart is empty, but we're submitting or received transaction response
+		if ( includes(
+			[ transactionStepTypes.SUBMITTING_WPCOM_REQUEST, transactionStepTypes.RECEIVED_WPCOM_RESPONSE ],
```

Comment:
Shouldn't happen. From my testing it's working ok. The problem is that we're returning `null` from the server on successful purchases (which we probably broke). :(